src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -457,20 +591,41 @@ class SegmentCleaner : public JournalSegmentProvider {
     return journal_head;
   }
 
-  void init_mark_segment_closed(segment_id_t segment, segment_seq_t seq) final {
+  void init_mark_segment_closed(
+    device_segment_t segment,
+    // number of segments of the current segment manager
+    segment_id_t num_segments_by_sm,
+    segment_seq_t seq,
+    bool out_of_line) final
+  {
     crimson::get_logger(ceph_subsys_seastore).debug(
       "SegmentCleaner::init_mark_segment_closed: segment {}, seq {}",
       segment,
       seq);
     mark_closed(segment);
+    auto& segment_manager = segments[segment.device_id()];
```

Comment:
I don't understand what's going on in this method.  Can we just initialize the the devices prior to scanning the segments?

Reply:
Sorry, this should have been removed...